react-native-scripts/src/util/packager.js

```diff
@@ -61,6 +62,40 @@ function run(onReady: () => ?any, options: Object = {}, isInteractive = false) {
   let progressBar;
   const projectDir = process.cwd();
 
+  if (process.platform !== 'win32') {
+    if (process.platform === 'darwin') {
+      const watcherDetails = spawn.sync('sysctl', ['kern.maxfiles']).stdout.toString();
+      if (parseInt(watcherDetails.split(':')[1].trim()) < 5242880) {
+        log.withTimestamp(
+          `
+The number of files that can be watched is too low. Please set it to higher number.
+Please use:
+${chalk.cyan(`  sudo sysctl -w kern.maxfiles=5242880
+  sudo sysctl -w kern.maxfilesperproc=524288`)}
+and start again.
+        `
+        );
+        process.exit(1);
+      }
+    } else {
+      const watcherDetails = spawn
+        .sync('sysctl', ['fs.inotify.max_user_watches'])
+        .stdout.toString();
+      if (parseInt(watcherDetails.split('=')[1].trim()) < 12288) {
```

Comment:
Where does this magic number come from? If I have very many existing watches, this might not be enough (and if I have very few, it might not be necessary to increase it). There's also not necessarily any correlation between the number of watches and inotify instances, so I think these should be checked and printed commands for separately.

I guess it would be harder to check, but to do this correctly I think you have to detect when the number of watches has reached it's limit, and then print the error.